src/python/pants/backend/python/target_types_rules.py

```diff
@@ -143,29 +159,174 @@ async def inject_pex_binary_entry_point_dependency(
 # -----------------------------------------------------------------------------------------------
 
 
+def _classify_entry_points(
+    entry_points: FrozenDict[str, FrozenDict[str, str]]
+) -> Generator[Tuple[bool, str, str, str], None, None]:
+    """Looks at each entry point to see if it is a target address or not.
+
+    Yields tuples: is_target, section, name, ref.
+    """
+    for section, values in entry_points.items():
+        for name, ref in values.items():
+            yield ref.startswith(":") or "/" in ref, section, name, ref
+
+
+@rule(desc="Determining the entry points for a `python_distribution` target", level=LogLevel.DEBUG)
+async def resolve_python_distribution_entry_points(
+    request: ResolvePythonDistributionEntryPointsRequest,
+) -> ResolvedPythonDistributionEntryPoints:
+    field_value = request.entry_points_field.value
+    if field_value is None:
+        return ResolvedPythonDistributionEntryPoints()
+
+    address = request.entry_points_field.address
+    classified_entry_points = list(_classify_entry_points(field_value))
+
+    # Pick out all target addresses up front, so we can use MultiGet later.
+    #
+    # This calls for a bit of trickery however (using the "y_by_x" mapping dicts), so we keep track
+    # of which address belongs to which entry point.
+    target_refs = [ref for is_target, _, _, ref in classified_entry_points if is_target]
+    # intermediate step, as Get(Targets) returns a deduplicated set..
+    target_addresses = await Get(
+        Addresses, UnparsedAddressInputs(target_refs, owning_address=address)
+    )
+    address_by_ref = dict(zip(target_refs, target_addresses))
+    targets = await Get(Targets, Addresses, target_addresses)
```

Comment:
I'm not following this part. Why not go right away to `await Get(Targets, UnparsedAddressInputs)`?

Reply:
Did you miss my comment, or maybe it didn't explain it well enough?
If I go right away to `Get(Targets)`, I don't have a 1:1 mapping from my inputs, in case of duplicates in the input, so can not know which unparsed address represents which target (which was the case in the test data, luckily!)

Reply:
I've added a few more words to that comment...